middle_end/flambda2/reaper/traverse.ml

```diff
@@ -0,0 +1,672 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*           Nathanaëlle Courant, Pierre Chambart, OCamlPro               *)
+(*                                                                        *)
+(*   Copyright 2024 OCamlPro SAS                                          *)
+(*   Copyright 2024 Jane Street Group LLC                                 *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+open! Flambda.Import
+open! Rev_expr
+open! Traverse_denv
+module Acc = Traverse_acc
+module Graph = Global_flow_graph
+module Dot = Dot_printer
+
+type denv = Traverse_denv.t
+
+type acc = Traverse_acc.t
+
+let apply_cont_deps denv acc apply_cont =
+  let cont = Apply_cont_expr.continuation apply_cont in
+  let args = Apply_cont_expr.args apply_cont in
+  let params = Continuation.Map.find cont denv.conts in
+  let (Normal params) = params in
+  List.iter2 (fun param dep -> Acc.cont_dep ~denv param dep acc) params args
+
+let prepare_code ~denv acc (code_id : Code_id.t) (code : Code.t) =
```

Comment:
Comment why some deps are not registered immediately: code has not been seen yet due to traversal order